index.js

```diff
@@ -5,125 +5,423 @@ class LiteApi {
+ *
+ * @returns {object} - The result of the operation.
+ *   - On success: { "status": "success", "data": <FullRatesResponse> }
+ *   - On failure: { "status": "failed", "error": "..." } or { "status": "failed", "errors": [ ... ] }
+ */
+async getFullRates(data, timeoutMs = 10000) {
+    // 1. Validate input
+    const errors = [];
+  
+    // Ensure `data` is an object
+    if (!data || typeof data !== 'object') {
+      errors.push('The `data` parameter is required and must be an object.');
+    } else {
+      // 1a. hotelIds
+      if (!Array.isArray(data.hotelIds) || data.hotelIds.length === 0) {
+        errors.push('`hotelIds` is required and must be a non-empty array of strings.');
+      }
+  
+      // 1b. checkin & checkout
+      if (!data.checkin || typeof data.checkin !== 'string') {
+        errors.push('`checkin` is required and must be a string in YYYY-MM-DD format.');
+      }
+      if (!data.checkout || typeof data.checkout !== 'string') {
+        errors.push('`checkout` is required and must be a string in YYYY-MM-DD format.');
+      }
+  
+      // 1c. currency
+      if (!data.currency || typeof data.currency !== 'string') {
+        errors.push('`currency` is required and must be a string (e.g., "USD").');
+      }
+  
+      // 1d. guestNationality
+      if (!data.guestNationality || typeof data.guestNationality !== 'string') {
+        errors.push('`guestNationality` is required and must be a string (e.g., "US").');
+      }
+  
+      // 1e. occupancies
+      if (!Array.isArray(data.occupancies) || data.occupancies.length === 0) {
+        errors.push('`occupancies` is required and must be a non-empty array.');
+      } else {
+        // Validate each occupancy object
+        data.occupancies.forEach((occupancy, index) => {
+          if (typeof occupancy !== 'object') {
+            errors.push(`occupancies[${index}] must be an object.`);
+          } else {
+            if (typeof occupancy.adults !== 'number') {
+              errors.push(`occupancies[${index}] must have a numeric 'adults' property.`);
             }
-        }
-
+            // If children are used, ensure it is an array of numbers
+            if (occupancy.children && !Array.isArray(occupancy.children)) {
+              errors.push(`occupancies[${index}].children must be an array of numbers if provided.`);
+            }
+          }
+        });
+      }
+    }
+  
+    // If any validation errors occurred, return them
+    if (errors.length > 0) {
+      return {
+        status: 'failed',
+        errors
+      };
+    }
+  
+    // 2. Prepare request options
+    const options = {
+      method: 'POST',
+      headers: {
+        accept: 'application/json',
+        'content-type': 'application/json',
+        'X-API-Key': this.apiKey
+      },
+      body: JSON.stringify(data)
+    };
+  
+    // 3. Setup AbortController for timeout
+    const controller = new AbortController();
```

Comment:
Where is this defined and what's this? @Achraf-El-khaier I think you're relying too much on AI.